kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/UnserializableComponentWrapper.java

```diff
@@ -106,16 +118,23 @@ private void writeObject(java.io.ObjectOutputStream out)
     private void readObject(java.io.ObjectInputStream in)
             throws IOException, ClassNotFoundException {
         in.defaultReadObject();
-        in.registerValidation(
-                () -> getUI().map(UI::getSession).ifPresent(session -> {
-                    Runnable cleaner = SessionUtil.injectLockIfNeeded(session);
-                    try {
-                        component = deserializer.apply(state);
-                        getElement().appendChild(component.getElement());
-                    } finally {
-                        cleaner.run();
-                    }
-                }), 0);
+        in.registerValidation(this::restoreComponent, 0);
+    }
+
+    private void restoreComponent() {
+        getUI().map(UI::getSession).ifPresent(session -> {
+            if (deserializer == null) {
+                throw new IllegalStateException(
+                        "Deserializer function not set");
+            }
```

Comment:
I would move the check in `readObject`, to make deserialization process fail fast